File: Crio/crio_programming_interview_problems-master/BinaryToString/BinaryToString.py

def convert_to_binary(number):
# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    if not (number*(2**31)).is_integer():
        return "ERROR"
    limit = number
    trailing_zeros = 0
    while not limit.is_integer():
        limit *= 2
        trailing_zeros += 1

    result = bin(int(limit))[::-1][:-2]
    return "0." + result.rjust(trailing_zeros, "0")
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def main():
    number = float(input())
    result = convert_to_binary(number)
    print(result)


if __name__ == "__main__":
    main()

File: Crio/crio_programming_interview_problems-master/BinaryToString/BinaryToString.java
import java.util.*;

public class BinaryToString {
    String convertToBinary(double number) {
        //CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        String answer = new String("0.");
        while(number > 0) {
            double temp = number*2;
            int integerPart = (int)temp;
            answer += Integer.toString(integerPart);
            number = number*2 - integerPart;
            if(answer.length() == 35) {
                answer = "ERROR";
                return answer;
            }
        }
        return answer;
        //CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
    }
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double number = scanner.nextDouble();
        String result = new BinaryToString().convertToBinary(number);
        System.out.println(result);
        scanner.close();
    }
}